Directory.Build.targets

```diff
@@ -20,4 +20,7 @@
     <NETCoreAppMaximumVersion>$(MajorVersion).$(MinorVersion)</NETCoreAppMaximumVersion>
   </PropertyGroup>
 
+  <Target Name="GeneratingFilesBeforeRunningSourceLink"
+          BeforeTargets="SetEmbeddedFilesFromSourceControlManagerUntrackedFiles"
+          DependsOnTargets="GenerateAssemblyInfo;GenerateInternalsVisibleToFile" />
```

Comment:
This shouldn't be necessary as long as all targets that add a `Compile` item have `BeforeTargets="BeforeCompile;CoreCompile"`.

Reply:
GenerateAssemblyInfo in the SDK had the issue though. You fixed it in dotnet/sdk#10613. Until we pick that SDK, a better workaround would be what was suggested in dotnet/sourcelink#572:
```xml
<Project>
  <PropertyGroup>
    <TargetFrameworkMonikerAssemblyAttributesPath>$([System.IO.Path]::Combine('$(IntermediateOutputPath)','$(TargetFrameworkMoniker).AssemblyAttributes$(DefaultLanguageSourceExtension)'))</TargetFrameworkMonikerAssemblyAttributesPath>
  </PropertyGroup>
  <ItemGroup>
    <EmbeddedFiles Include="$(GeneratedAssemblyInfoFile)"/>
  </ItemGroup>
</Project>
```